### PR TITLE
feat(dashmate): implement http json rpc api

### PR DESCRIPTION
**Pull request by pshenmic**
## Issue being fixed or feature implemented

JSON RPC api wrapper around dashmate enables programmatic invocation of dashmate commands. It allows you remotely execute commands and fetch node status. Developers will be able to build an UI on top of it.

## What was done?

* Added JSON RPC api

## How Has This Been Tested?

Locally

## Breaking Changes

No

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by pshenmic** [comment]
888 😈😈

File: packages/dashmate/configs/schema/configJsonSchema.js [hunk 1]
```diff
@@ -669,6 +669,19 @@ module.exports = {
             docker: {
               $ref: '#/definitions/docker',
             },
+            jsonRpc: {
```
Comment (shumkov) [inline packages/dashmate/configs/schema/configJsonSchema.js:672]: Let's rename to API

File: packages/dashmate/configs/schema/configJsonSchema.js [hunk 2]
```diff
@@ -669,6 +669,19 @@ module.exports = {
...
+                },
+                port: {
+                  $ref: '#/definitions/port',
+                },
+              },
+              required: ['enabled', 'port'],
+              additionalProperties: false,
+            },
           },
           required: ['docker'],
```
Comment (shumkov) [inline packages/dashmate/configs/schema/configJsonSchema.js:686]: It must be here as well

File: packages/dashmate/configs/system/local.js [hunk 3]
```diff
@@ -48,6 +48,13 @@ module.exports = lodashMerge({}, baseConfig, {
       },
     },
   },
+  dashmate: {
+    helper: {
+      jsonRpc: {
```
Comment (shumkov) [inline packages/dashmate/configs/system/local.js:53]: api

File: packages/dashmate/scripts/helper.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ const dotenv = require('dotenv');
 const { asValue } = require('awilix');
 const createDIContainer = require('../src/createDIContainer');
 
-(async function main() {
+const httpApiFactory = require('../src/helper/api/httpApiFactory')(async function main() {
```
Comment (shumkov) [inline packages/dashmate/scripts/helper.js:5]: It looks super unreadable. You should use container to resolve HTTP server

File: packages/dashmate/scripts/helper.js [hunk 5]
```diff
@@ -43,4 +46,7 @@ const createDIContainer = require('../src/createDIContainer');
     // prevent infinite restarts
     setInterval(() => {}, 60 * 1000);
   }
+
+  const httpApi = container.resolve('httpApi');
+  httpApi(container);
```
Comment (shumkov) [inline packages/dashmate/scripts/helper.js:51]: You can specify `container` inside as a dependency if you need

File: packages/dashmate/src/helper/api/httpApiFactory.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
+const jayson = require('jayson/promise');
+const oclif = require('@oclif/core');
+
+/**
+ *
+ * @return {httpApi}
+ */
+function httpApiFactory() {
```
Comment (shumkov) [inline packages/dashmate/src/helper/api/httpApiFactory.js:8]: you should put container dependencies here

File: packages/dashmate/src/listr/tasks/setup/setupLocalPresetTaskFactory.js [hunk 7]
```diff
@@ -244,6 +244,8 @@ function setupLocalPresetTaskFactory(
                     masternodeRewardSharesDerivedSecondPrivateKey.privateKey
                       .toPublicKey().toString(),
                   );
+
+                  config.set('dashmate.helper.jsonRpc.port', config.get('dashmate.helper.jsonRpc.port') + (i * 100));
```
Comment (shumkov) [inline packages/dashmate/src/listr/tasks/setup/setupLocalPresetTaskFactory.js:248]: You open port for each config?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/dashmate/src/listr/tasks/setup/setupLocalPresetTaskFactory.js (hunk 7 above)
Comment (pshenmic) [reply packages/dashmate/src/listr/tasks/setup/setupLocalPresetTaskFactory.js:248]: Yes

**Review by shumkov — APPROVED** [review]
👍

**Review by shumkov — APPROVED** [review]

**Pull request merged**